tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/ProcessorImpl.cpp

```diff
@@ -0,0 +1,520 @@
+      // However if the type is in the _internal namespace, then we want to exclude it if
+      // we're excluding internal types.
+      if ((typeName.find("::_internal") != std::string::npos)
+          && !m_processorImpl->IncludeInternal())
+      {
+        shouldCollect = false;
+      }
+      // However if the type is in the _detail namespace, then we want to exclude it if we're
+      // excluding detail types.
+      if ((typeName.find("::_detail") != std::string::npos) && !m_processorImpl->IncludeDetail())
+      {
+        shouldCollect = false;
+      }
+    }
+    else
+    {
+      // We have a namespace filter set. Look at all types whose names start with the filter
+      // namespace name. We don't want to consider namespaces, since they're not descrete entries in
+      // our resulting output.
+      const std::string typeName{namedDecl->getQualifiedNameAsString()};
+      if (namedDecl->getKind() != Decl::Namespace)
+      {
+        if (std::find_if(
+                std::begin(m_processorImpl->FilterNamespaces()),
+                std::end(m_processorImpl->FilterNamespaces()),
+                [&typeName](std::string const& ns) { return typeName.find(ns) == 0; })
+            == std::end(m_processorImpl->FilterNamespaces()))
+        {
+          shouldCollect = true;
+          bool generateError = true;
+          // Don't flag "using" declarations or forward declarations.
+          if ((namedDecl->getKind() == Decl::Using) || (namedDecl->getKind() == Decl::TypeAlias)
+              || (namedDecl->getKind() == Decl::CXXRecord
+                  && cast<CXXRecordDecl>(namedDecl)->getDefinition() != namedDecl))
+          {
+            generateError = false;
+          }
+          if (generateError)
+          {
+            m_processorImpl->GetClassesDatabase()->CreateApiViewMessage(
+                ApiViewMessages::TypeDeclaredInNamespaceOutsideFilter, typeName);
+          }
+        }
+      }
+      if (shouldCollect)
+      {
+        // However if the type is in the _internal namespace, then we want to exclude it if we're
+        // excluding internal types.
+        if (typeName.find("::_internal") != std::string::npos)
+        {
+          if (!m_processorImpl->IncludeInternal())
+          {
+            shouldCollect = false;
+          }
+          // If this _internal type is not in a namespace starting with "Azure::Core", then
+          // we want to flag it as an error and include the type.
+          if (typeName.find("Azure::Core") != 0)
+          {
+            m_processorImpl->GetClassesDatabase()->CreateApiViewMessage(
+                ApiViewMessages::InternalTypesInNonCorePackage, typeName);
+            shouldCollect = true;
+          }
+        }
+        // If the type is in the _detail namespace, then we want to exclude it if we're
+        // excluding detail types.
+        if ((typeName.find("::_detail") != std::string::npos) && !m_processorImpl->IncludeDetail())
```

Comment:
We should flag items in the `namespcae _detail {`.  i.e. anything that isn't under `Azure::` but is within `_detail`